plugins/module_utils/network/awplus/config/policy_interfaces/policy_interfaces.py

```diff
@@ -0,0 +1,234 @@
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = []
+        want = [] if not want else want
+        w_ints = [w_pol_int.get('int_name') for w_pol_int in want]
+        for h_pol_int in have:
+            h_int_name = h_pol_int.get('int_name')
+            if h_int_name in w_ints:
+                for w_pol_int in want:
+                    w_int_name = w_pol_int.get('int_name')
+                    if w_int_name == h_int_name:
+                        # update entry in have to match want
+                        commands.extend(self.change_config(w_pol_int, h_pol_int, replace=True))
+            else:
+                # have entry does not exist in want so delete
+                commands.extend(self.change_config(h_pol_int, h_pol_int, delete=True))
+        return commands
+
+    @staticmethod
+    def _state_merged(self, want, have):
+        """ The command generator when state is merged
+
+        :rtype: A list
+        :returns: the commands necessary to merge the provided into
+                  the current configuration
+        """
+        commands = []
+        want = [] if not want else want
+        for h_pol_int in have:
+            h_int_name = h_pol_int.get('int_name')
+            for w_pol_int in want:
+                w_int_name = w_pol_int.get('int_name')
+                if w_int_name == h_int_name:
+                    commands.extend(self.change_config(w_pol_int, h_pol_int, merge=True))
+        return commands
+
+    @staticmethod
+    def _state_deleted(self, want, have):
+        """ The command generator when state is deleted
+
+        :rtype: A list
+        :returns: the commands necessary to remove the current configuration
+                  of the provided objects
+        """
+        commands = []
+        want = [] if not want else want
+        for h_pol_int in have:
+            h_int_name = h_pol_int.get('int_name')
+            for w_pol_int in want:
+                w_int_name = w_pol_int.get('int_name')
+                if w_int_name == h_int_name:
+                    commands.extend(self.change_config(w_pol_int, h_pol_int, delete=True))
+        return commands
+
+    def change_config(self, w_pol_int, h_pol_int, delete=False, merge=False, replace=False):
+        """ The command generator when state is deleted
+        :param w_pol_int: dictionary of the policy map and the interface to apply to
+        :param h_pol_int: dictionary of the current configuration of the interface in w_pol_int
+        :param delete: flag to specify a delete action
+        :param merge: flag to specify a merge action
+        :param replace: flag to specify a replace action
+        :rtype: A list
+        :returns: the commands necessary to remove the current configuration
+                  of the provided objects
+        """
+        cmd = []
+        int_name = h_pol_int.get('int_name')
```

Comment:
logic from here on down is very complex. I think it would be good to try to make it a bit clearer or add some comment. Really terse code can be very efficient, but it doesn't seem very maintainable.